include/pybind11/cast.h

```diff
@@ -1631,6 +1631,14 @@ struct pyobject_caster {
 
     template <typename T = type, enable_if_t<std::is_base_of<object, T>::value, int> = 0>
     bool load(handle src, bool /* convert */) {
+#ifndef PYBIND11_DISABLE_IMPLICIT_STR_FROM_BYTES
```

Comment:
JK, why does this have an escape hatch? It seems to not have been added as an option to the CMakeLists. (an addition which should wait till after #2370 if it gets added, anyway)

Reply:
Hmmm, wait, this isn't mine. This is coming from @rwgk's #2380 (which I had to rebase on to get some tests working, at some point). Let me check this again (and probably rebase onto the latest `str-bytes-cleanup`, though I thought I'd done so...).

Reply:
OK, fixed. Not sure what went wrong, but thanks for catching that!

Reply:
Full disclosure on this macro: https://github.com/pybind/pybind11/pull/2380#discussion_r469575686
So, as far as I know, it's not supposed to be public, and it's still up for debate on what to do with ti :-)